test/unit/utils/sql.test.ts

```diff
@@ -264,3 +264,327 @@ describe('mapBindParameters', () => {
     });
   });
 });
+
+describe('injectReplacements (named replacements)', () => {
+  it('parses named replacements', () => {
+    const sql = injectReplacements(`SELECT ${dialect.TICK_CHAR_LEFT}:id${dialect.TICK_CHAR_RIGHT} FROM users WHERE id = ':id' OR id = :id OR id = ''':id'''`, dialect, {
+      id: 1,
+    });
+
+    expectsql(sql, {
+      default: `SELECT [:id] FROM users WHERE id = ':id' OR id = 1 OR id = ''':id'''`,
+    });
+  });
+
+  it('throws if a named replacement is not provided as an own property', () => {
+    expect(() => {
+      injectReplacements(`SELECT * FROM users WHERE id = :toString`, dialect, {
+        id: 1,
+      });
+    }).to.throw('Named replacement ":toString" has no entry in the replacement map.');
+
+  });
+
+  it('parses named replacements followed by cast syntax', () => {
+    const sql = injectReplacements(`SELECT * FROM users WHERE id = :id::string`, dialect, {
+      id: 1,
+    });
+
+    // this is postgres-only syntax. It doesn't work in other dialects.
+    expectsql(sql, {
+      default: `SELECT * FROM users WHERE id = 1::string`,
+    });
+  });
+
+  it('parses single letter named replacements', () => {
+    const sql = injectReplacements(`SELECT * FROM users WHERE id = :a`, dialect, {
+      a: 1,
+    });
+
+    expectsql(sql, {
+      default: `SELECT * FROM users WHERE id = 1`,
+    });
+  });
+
+  it(`does not consider the token to be a replacement if it does not follow '(', ',', '=' or whitespace`, () => {
+    const sql = injectReplacements(`SELECT * FROM users WHERE id = fn(:id) OR id = fn('a',:id) OR id=:id`, dialect, {
```

Comment:
Do we want to add `OR id = :id` here because the test name mentions whitespace?

Reply:
Ah yes

Reply:
Done